### PR TITLE
Add the option to emit events when an authorization request is sent

### PR DESCRIPTION
**Pull request by xavery**

### CODE REVIEW

File: src/plugin/ui.c [hunk 1]
```diff
@@ -102,6 +102,12 @@ purple_events_ui_get_pref_frame(PurplePlugin *plugin)
     purple_plugin_pref_frame_add(frame, pref);
 
     pref = purple_plugin_pref_new_with_name_and_label(
+        "/plugins/core/events/events/authorization-requested",
+        _("Buddy requests authorization")
```
Comment (sardemff7) [inline src/plugin/ui.c]: Please keep the -ing form here

File: src/plugin/events.c [hunk 2]
```diff
@@ -138,6 +139,8 @@ _purple_events_init(PurplePlugin *plugin)
                            purple_value_new(PURPLE_TYPE_STRING), purple_value_new(PURPLE_TYPE_STRING), purple_value_new(PURPLE_TYPE_STRING), purple_value_new(PURPLE_TYPE_STRING));
     purple_signal_register(plugin, "conversation-got-focus", purple_marshal_VOID__POINTER_POINTER_POINTER_POINTER, NULL, 1,
                            purple_value_new(PURPLE_TYPE_SUBTYPE, PURPLE_SUBTYPE_CONVERSATION));
+    purple_signal_register(plugin, "authorization-requested", purple_marshal_VOID__POINTER_POINTER_POINTER, NULL, 3,
```
Comment (sardemff7) [inline src/plugin/events.c]: The event is specific to a buddy, so it must be prefixed with `user_` as the others, and sorted before the `conversation-got-focus` one, which is an internal ones to dismiss notifications.

File: src/plugin/callbacks.c [hunk 3]
```diff
@@ -226,3 +226,12 @@ purple_events_callback_account_signed_on(PurpleConnection *conn, PurpleEventsCon
 
     data->handle = purple_timeout_add_seconds(5, _purple_events_callback_account_signed_on_timeout, data);
 }
+
+void
+purple_events_callback_auth_requested(PurpleAccount *account, const char *remote_user, const char *message, PurpleEventsContext *context)
```
Comment (sardemff7) [inline src/plugin/callbacks.c]: Please follow the style of others callbacks. `remote_user` should be `sender`.

File: src/plugin/callbacks.c [hunk 4]
```diff
@@ -226,3 +226,12 @@ purple_events_callback_account_signed_on(PurpleConnection *conn, PurpleEventsCon
...
+{
+    if ( ! purple_events_utils_check_event_dispatch(context, account, NULL, "authorization-requested") )
```
Comment (sardemff7) [inline src/plugin/callbacks.c]: The event has a sender, please do as the other events and check `purple_events_utils_check_buddy_event_dispatch()` too.

Comment (xavery) [reply src/plugin/callbacks.c]: It has a sender, but not a buddy : at the time this event arrives, the sender is not yet in the receiver's buddy list and thus has no associated `PurpleBuddy` structure in the `PurpleAccount`.

**Comment by xavery** [comment]
Thanks for the review, I pushed the requested changes. Let me know what to do about `purple_events_utils_check_buddy_event_dispatch`, since it doesn't seem doable to me in this particular situation.

File: src/plugin/callbacks.c (hunk 4 above)
Comment (sardemff7) [reply src/plugin/callbacks.c]: Fair enough.
Are there protocols needing an authorization ping pong? Like, you add someone, they authorize you and add you, then you have to authorize them back? You probably won’t get the time to block the event anyway… Let’s keep it simple.

Comment (xavery) [reply src/plugin/callbacks.c]: The only protocol plugins present in the libpurple codebase that use `purple_account_request_authorization` are Jabber and OSCAR. None of those use ping-pong authorisation. Personally, my own motivation for these changes was the skypeweb plugin, which doesn't use ping-pong either. I think it's safe to assume that such a scheme isn't used nowadays...

**Pull request merged**